benchmark/camelyon/common/dataset_manager.py

```diff
@@ -1,11 +1,52 @@
 import os
 import shutil
+import subprocess
 from pathlib import Path
 
 import numpy as np
+from tqdm.auto import tqdm
 
 ROOT_DIR = Path(__file__).parents[1]
 
+FILE_LIST = [
+    "normal_001.tif.npy",
+    "normal_002.tif.npy",
+    "tumor_106.tif.npy",
+    "tumor_108.tif.npy",
+]
+URL = "https://zenodo.org/api/files/d5520b8a-6a5b-41ef-bde0-247dbdded101/"
+N_PARALLEL = 4
+TIMEOUT = 600
+
+
+def fetch_camelyon(data_path: Path):
+    index_filename = "index.csv"
+    if not (data_path / index_filename).exists():
+        print("Downloading the dataset (~400Mb), this may take a few minutes.")
+        subprocess.run(
+            ["wget", f"{URL}index.csv"],
+            check=True,
+            cwd=data_path,
+        )
+        processes = [
+            subprocess.Popen(
+                ["wget", f"{URL}{filename}"],
+                cwd=data_path / "tiles_0.5mpp",
+            )
+            for filename in FILE_LIST
+        ]
+        errors = list()
+        for proc in tqdm(processes):
+            try:
+                proc.communicate(timeout=TIMEOUT)
+            except subprocess.TimeoutExpired:
+                proc.kill()
+                proc.communicate(timeout=TIMEOUT)
+            if proc.returncode != 0:
+                errors.append((proc.args, proc.returncode))
+        if len(errors) > 0:
+            raise subprocess.CalledProcessError(proc.args, proc.returncode)
```

Comment:
Here we display only the last `proc` error, not the whole `error` list, is it wanted?